datafusion/sql/src/utils.rs

```diff
@@ -246,3 +246,168 @@ pub(crate) fn normalize_ident(id: Ident) -> String {
         None => id.value.to_ascii_lowercase(),
     }
 }
+
+pub(crate) fn unescape(s: &str) -> Option<String> {
+    let mut queue: VecDeque<_> = String::from(s).chars().collect();
```

Comment:
I *think* you might be able to just get away with [`Chars`](https://doc.rust-lang.org/std/str/struct.Chars.html) instead of a `VecDeque`, as you're only popping from the front anyway, and `Chars` allows you to iterate from the front and also peek. Might be worth exploring towards :thinking: